crypto/nextgen_crypto/src/bls12381.rs

```diff
@@ -106,10 +106,24 @@ impl Uniform for BLS12381PrivateKey {
     where
         R: ::rand::SeedableRng + ::rand::RngCore + ::rand::CryptoRng,
     {
-        let mut fr_repr: [u64; 4usize] = rng.gen();
-        // Since field modulus is 381-bit prime, drop the 3 highest-order bits
-        fr_repr[3] &= 0x1FFF_FFFF_FFFF_FFFF;
-        let mut fr = Fr::from_repr(FrRepr(fr_repr)).unwrap();
+        let mut fr;
+        loop {
+            let mut fr_repr: [u64; 4usize] = rng.gen();
+            fr_repr[3] &= 0xFFFF_FFFF_FFFF_FFFF >> 1;
+
+            // stop sampling if the resulting number is less than the modulus r
+            if FrRepr(fr_repr)
+                < FrRepr([
+                    0xffff_ffff_0000_0001,
```

Comment:
Better create a private function that returns r and invoke it here, so it’s more clear what this number is. Add a link to BLS 12 381 specs rfc (or any other official page) to this function’s doc comment.

Reply:
:+1: to documenting magic numbers

Reply:
Thank you for your suggestions! All very useful, I've amended the commit with introducing a constant, changed the docs referring to the spec.